src/material.cpp

```diff
@@ -353,6 +353,35 @@ Material::~Material()
   model::material_map.erase(id_);
 }
 
+Material & Material::clone()
+{
+  std::unique_ptr<Material> mat = std::make_unique<Material>();
+  mat->index_ = model::materials.size();
+  mat->set_id(C_NONE);
+
+  // set all other parameters to whatever the calling Material has
+  mat->name_ = name_;
+  mat->nuclide_ = nuclide_;
+  mat->element_ = element_;
+  mat->ncrystal_mat_ = ncrystal_mat_;
```

Comment:
This is going to create a copy of the pointer to an NCrystal material. I won't let it hold up this PR, but @marquezj do you know if we should also be creating a new instance of an NCrystal material here?

Reply:
Thanks @pshriwise. Better hold it until @tkittel see this. 

NCrystalMat is a wrapper that contains the NCrystal configuration string and a shared pointer to the actual NCrystal material. Although the idea of clone is to deep-copy, neither the NCrystal configuration string or the material are meant to be modified in execution, so maybe it is fine to just copy the pointer. What do you think @tkittel?

(Which reminds me: we should check that materials that come from NCrystal are non-depletable, at least for now)

Reply:
Indeed, it should be safe. The NCrystalMat class just has two data-members which are essentially immutable: One is a `std::string` and the other a `std::shared_ptr<const NCrystal::ProcImpl::Process>`. The `NCrystal::ProcImpl::Process` interface only exposes const methods.

So for practical purposes it can be considered both cheap and safe to clone `NCrystalMat` objects via a simple copy.